### PR TITLE
Feature/tm/basic telemetry manager mavlink interpreter

### PR DESCRIPTION
**Pull request by Yaremadzulynsky**
## Description
 ** THIS IS A REBUILD OF A PREVIOUS PR **
*What was completed, changed, or updated?* 
Added official Mavlink library.

<br/>

*Why was this done (if applicable)?*

N/A

<br/>

## Testing

*What **manual** tests were used to validate the code?*

None

<br/>

*What **unit** tests were used to validate the code?*

None

<br/>

## Documentation

*Milestone number and name:*

*Link to Asana task:*

*Link to Confluence documentation:*

<br/>

## Reminders

- [x] Add reviewers to the PR

- [x] Mention the PR in the appropriate discord channel

### CODE REVIEW

**Comment by Yaremadzulynsky** [comment]
I previously submitted a PR for this feature but struggled to get it approved. I spoke to Aidan and he mentioned that it could be due to the massive amount of files to review. He suggested I do one commit for the official Mavlink library, then once it is approved into the main branch to do a PR with the code I wrote. This commit is only the Mavlink library.

File: Tools/Firmware/CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +41,10 @@ include_directories(
     ${ROOT_DIR}/Boardfiles/${FOLDER_NAME}/Middlewares/Third_Party/FreeRTOS/Source/CMSIS_RTOS_V2
     ${ROOT_DIR}/Boardfiles/${FOLDER_NAME}/Middlewares/Third_Party/FreeRTOS/Source/portable/GCC/ARM_${PORTABLE_NAME}
     ${ROOT_DIR}/Boardfiles/${FOLDER_NAME}/Middlewares/Third_Party/FreeRTOS/Source/include
+    
+    ## Models Includes ##
+    ${ROOT_DIR}/Models
+    ${ROOT_DIR}/Models/${MODEL_NAME}
```
Comment (SnackkOverflowError) [inline Tools/Firmware/CMakeLists.txt:47]: whats going on here

File: Tools/Firmware/CMakeLists.txt [hunk 2]
```diff
@@ -28,6 +28,9 @@ include_directories(
     ${ROOT_DIR}/Drivers/ccontrol/Inc
     ${ROOT_DIR}/Drivers/sensor_fusion/Inc
     ${ROOT_DIR}/Drivers/IWDG_Driver/Inc
+
+    ${ROOT_DIR}/Drivers/MotorChannel/Inc
```
Comment (SnackkOverflowError) [inline Tools/Firmware/CMakeLists.txt:32]: is this needed for TM?

File: TelemetryManager/Inc/c_library_v2/cubepilot/cubepilot.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/** @file
+ *  @brief MAVLink comm protocol generated from cubepilot.xml
+ *  @see http://mavlink.org
```
Comment (SnackkOverflowError) [inline TelemetryManager/Inc/c_library_v2/cubepilot/cubepilot.h:3]: Do we need the entire library or can we make do with a subset? for example do we need this cubepilot folder?

**Review by SnackkOverflowError — COMMENTED** [review]
some questions/nitpicks

File: TelemetryManager/Inc/c_library_v2/cubepilot/cubepilot.h (hunk 3 above)
Comment (Yaremadzulynsky) [reply TelemetryManager/Inc/c_library_v2/cubepilot/cubepilot.h:3]: Good point. Will look into it. Likely not.

File: Tools/Firmware/CMakeLists.txt (hunk 1 above)
Comment (Yaremadzulynsky) [reply Tools/Firmware/CMakeLists.txt:47]: Not sure. I copied and pasted the file contents from the current main branch.

File: Tools/Firmware/CMakeLists.txt (hunk 2 above)
Comment (Yaremadzulynsky) [reply Tools/Firmware/CMakeLists.txt:32]: I don't think so. I copied and pasted the file contents from the current main branch.

File: TelemetryManager/Inc/c_library_v2/cubepilot/cubepilot.h (hunk 3 above)
Comment (Yaremadzulynsky) [reply TelemetryManager/Inc/c_library_v2/cubepilot/cubepilot.h]: I have removed the unnecessary folders from the library.

File: Tools/Firmware/CMakeLists.txt (hunk 1 above)
Comment (Yaremadzulynsky) [reply Tools/Firmware/CMakeLists.txt]: Am I good to resolve this conversation?

File: Tools/Firmware/CMakeLists.txt (hunk 2 above)
Comment (Yaremadzulynsky) [reply Tools/Firmware/CMakeLists.txt:32]: Am I good to resolve this conversation?

File: Tools/Firmware/CMakeLists.txt (hunk 1 above)
Comment (billylin609) [reply Tools/Firmware/CMakeLists.txt]: This is the Model Config code already in main.

File: Tools/Firmware/CMakeLists.txt (hunk 2 above)
Comment (billylin609) [reply Tools/Firmware/CMakeLists.txt:32]: PWM motor control code already in main

File: TelemetryManager/Inc/c_library_v2/checksum.h [hunk 4]
```diff
@@ -0,0 +1,95 @@
+#pragma once
```
Comment (SnackkOverflowError) [inline TelemetryManager/Inc/c_library_v2/checksum.h:1]: can we rename this folder to like something mavlink related, "c_library_v2" isnt very descriptive for our use case

**Review by SnackkOverflowError — CHANGES_REQUESTED** [review]

File: TelemetryManager/Inc/c_library_v2/checksum.h (hunk 4 above)
Comment (Yaremadzulynsky) [reply TelemetryManager/Inc/c_library_v2/checksum.h:1]: Right. That would make much more sense. Will do.

Comment (Yaremadzulynsky) [reply TelemetryManager/Inc/c_library_v2/checksum.h]: Done ✅

**Review by billylin609 — APPROVED** [review]
looks good pr approved

**Review by SnackkOverflowError — APPROVED** [review]

**Pull request merged**